### PR TITLE
Add GitHub Actions CI configuration

### PR DESCRIPTION
**Pull request by kjsanger**

### CODE REVIEW

File: .github/workflows/run-tests.yml [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+      - name: "Install Singularity"
+        run: |
+          pkg="singularity-ce_${SINGULARITY_VERSION}-$(lsb_release -sc)_amd64.deb"
+          curl -sSL -O "https://github.com/sylabs/singularity/releases/download/v${SINGULARITY_VERSION}/$pkg"
+          curl -sSL -O "https://github.com/sylabs/singularity/releases/download/v${SINGULARITY_VERSION}/sha256sums"
+          grep "$os_release" sha256sums | sha256sum
+          sudo apt-get install -q -y "./$pkg"
+          
+          # Avoid the runner's loop devices being exhausted
+          sudo singularity config global --set "max loop devices" 128
```
Comment (jmtcsngr) [inline .github/workflows/run-tests.yml]: Assuming this is fixed in singularity 4+, do we want to keep the manual setting?

File: .github/workflows/run-tests.yml [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+
+          # Enable DBus for this user, so that rootless cgroups works and we can create
+          # Singularity service instances. See https://github.com/containers/podman/issues/5443
+          systemctl --user enable --now dbus.socket
+          echo DBUS_SESSION_BUS_ADDRESS="unix:path=/run/user/$(id -u)/bus" >> $GITHUB_ENV
+
+      - name: "Cache Singularity images"
+        uses: actions/cache@v3
```
Comment (jmtcsngr) [inline .github/workflows/run-tests.yml]: I'm getting a warning from github the action should be updated. Dependency of the dependency update. There is a 2 week old release for the action so I thought worth testing. It worked for me.
```suggestion
        uses: actions/cache@v4
```

File: .github/workflows/run-tests.yml (hunk 1 above)
Comment (kjsanger) [reply .github/workflows/run-tests.yml]: It's probably not necessary, but I wanted a run to pass with it in place first. I'll try without

Comment (jmtcsngr) [reply .github/workflows/run-tests.yml]: Tried a few times with the change myself, and things were happy. But we can keep it until we are sure. Will approve the PR and you make the call when merging.

**Review by jmtcsngr — APPROVED** [review]

**Pull request merged**